regions/shapes/annulus.py

```diff
@@ -112,3 +111,329 @@ def to_pixel(self, wcs):
+    def center(self):
+        return self.region1.center
+
+    @property
+    def inner_width(self):
+        return self.region1.width
+
+    @property
+    def inner_height(self):
+        return self.region1.height
+
+    @property
+    def outer_width(self):
+        return self.region2.width
+
+    @property
+    def outer_height(self):
+        return self.region2.height
+
+    @property
+    def angle(self):
+        return self.region2.angle
+
+    def to_pixel(self, wcs):
+        center, scale, north_angle = skycoord_to_pixel_scale_angle(self.center, wcs)
+        # FIXME: The following line is needed to get a scalar PixCoord
+        center = PixCoord(float(center.x), float(center.y))
+
+        inner_width = self.inner_width.to('deg').value * scale
+        inner_height = self.inner_height.to('deg').value * scale
+        outer_width = self.outer_width.to('deg').value * scale
+        outer_height = self.outer_height.to('deg').value * scale
+        angle = self.angle + (north_angle - 90 * u.deg)
+
+        return EllipseAnnulusPixelRegion(center, inner_width, inner_height,
+                                         outer_width, outer_height, angle, self.meta, self.visual)
+
+
+class RectangleAnnulusPixelRegion(CompoundPixelRegion):
+    """
+    A rectangular annulus in pixel coordinates.
+
+    Parameters
+    ----------
+    center: `~regions.PixCoord`
+        The position of the center of the rectangular annulus.
+    inner_width: float
+        The inner width of the rectangular annulus (before rotation) in pixels
+    inner_height: float
+        The inner height of the rectangular annulus (before rotation) in pixels
+    outer_width: float
+        The outer width of the rectangular annulus (before rotation) in pixels
+    outer_height: float
+        The outer height of the rectangular annulus (before rotation) in pixels
+    angle: `~astropy.units.Quantity`
+        The rotation angle of the rectangular annulus, measured anti-clockwise.
+        If set to zero (the default), the width axis is lined up with the x axis.
+    """
+
+    def __init__(self, center, inner_width, inner_height, outer_width,
+                 outer_height, angle, meta=None, visual=None):
+        if inner_width > outer_width:
+            raise ValueError('Outer width should be larger than inner width.')
+
+        if inner_height > outer_height:
+            raise ValueError('Outer height should be larger than inner height.')
+
+        region1 = RectanglePixelRegion(center, inner_width, inner_height, angle)
+        region2 = RectanglePixelRegion(center, outer_width, outer_height, angle)
+
+        super(RectangleAnnulusPixelRegion, self).__init__(
+            region1=region1, region2=region2, operator=operator.xor, meta=meta, visual=visual)
+
+        self._repr_params = ('inner_width', 'inner_height',
+                             'outer_width', 'outer_height')
+
+    @property
+    def center(self):
+        return self.region1.center
```

Comment:
This is the same as for the ellipticalannulus, right?  Can we abstract them out to a metaclass?

Reply:
I also thought about this , but then again I thought why complicate the hierarchy for the sake of a few lines since that would be illogical, IMHO.

Reply:
It's not really a few lines, more like ~20-40 per region type, and it's repeated code.  I don't think it complicates the hierarchy much either.  I'm much more concerned about repeated code than about complicated inheritance, especially since the set of regions we have now may not be complete.

Reply:
Hmm.. you are right.. But, the elliptical annulus and rectangular annulus share a lot of repeated code but not circular annulus, how do you suggest they have a meta class? 
Or may we can have a `EllipseAnnulusRegion` parent meta class for `EllipseAnnulusPixelRegion` and `EllipseAnnulusSkyRegion`?
What do you say?

Reply:
How about `AsymmetricAnnulus` as the parent class for both rectangle & ellipse annulus types?  That would capture width, height, center, angle, area for both, right?  Thought, area would have to be overloaded.

Reply:
Seems nice. let's try that!